internal/pkg/gateway/api.go

```diff
@@ -427,11 +428,81 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 		return nil, status.Errorf(codes.Unavailable, "no orderer nodes available")
 	}
 
+	logger := logger.With("txID", request.TransactionId)
+	config := gs.getChannelConfig(request.ChannelId)
+	if config.ChannelConfig().Capabilities().ConsensusTypeBFT() {
+		return gs.submitBFT(ctx, orderers, txn, config, logger)
+	} else {
+		return gs.submitNonBFT(ctx, orderers, txn, logger)
+	}
+}
+
+func (gs *Server) submitBFT(ctx context.Context, orderers []*orderer, txn *common.Envelope, config channelconfig.Resources, logger *flogging.FabricLogger) (*gp.SubmitResponse, error) {
+	// For BFT, we send transaction to ALL orderers
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	waitCh := make(chan *gp.ErrorDetail, len(orderers))
+	for _, o := range orderers {
+		go func(ord *orderer) {
+			logger.Infow("Sending transaction to orderer", "endpoint", ord.logAddress)
+			ctx, cancel := context.WithTimeout(ctx, gs.options.BroadcastTimeout)
+			defer cancel()
```

Comment:
(Placing the comment here randomly)

Can we perhaps abstract this method into a function, and actually make a unit test for this? This method seems complex enough to deep a unit test of its own.